app/controllers/api/base_controller.rb

```diff
@@ -37,6 +37,24 @@ def verify_mfa_requirement
     render plain: "Gem requires MFA enabled; You do not have MFA enabled yet.", status: :forbidden
   end
 
+  def response_with_warnings(response)
+    response_with_mfa_warning(response)
+  end
+
+  def response_with_mfa_warning(response)
+    message = response
+    if @api_key.user.mfa_recommended_not_yet_enabled?
+      message += "\n\n[WARNING] For protection of your account and gems, we encourage you to set up multifactor authentication " \
```

Comment:
[Squiggly heredoc](https://infinum.com/blog/multiline-strings-ruby-2-3-0-the-squiggly-heredoc/) might be nice here (and for other multiline strings in the PR).

Reply:
I was considering using a HEREDOC. However, I intended the string (and others) to be a single line string because terminal sizes could mess formatting up. Now looking at rubygems/bundler, the convention appears to leave long messages in a single string instead of making it multiline ([ex 1](https://github.com/rubygems/rubygems/blob/9aeff4e4293d87c06e2469616f38cbcfc99bdc04/bundler/exe/bundle#L23-L27), [ex 2](https://github.com/rubygems/rubygems/blob/112b3b923f9be83663fe1cb32cceac413c4d9379/bundler/spec/bundler/dsl_spec.rb#L290-L293)).

Reply:
Oh! Of course, that makes sense.

Tangentially: Github needs a forehead-slap emoji.

Reply:
forehead-slap emoji me. TIL Tom told me that you can escape newlines by having a `\` at the end of a heredoc. So we can have a single line string. It looks a lot better with a HEREDOC than having 2 `\n` at the start.